packages/simcore-sdk/tests/integration/test_node_ports_aws_s3_cli.py

```diff
@@ -0,0 +1,417 @@
+# pylint: disable=redefined-outer-name
+# pylint: disable=unused-argument
+
+import asyncio
+import filecmp
+import os
+import urllib.parse
+from collections.abc import AsyncIterator, Callable
+from pathlib import Path
+from typing import Final
+from unittest.mock import AsyncMock
+from uuid import uuid4
+
+import aioboto3
+import aiofiles
+import pytest
+from faker import Faker
+from models_library.progress_bar import ProgressReport
+from pydantic import AnyUrl, ByteSize, parse_obj_as
+from servicelib.file_utils import remove_directory
+from servicelib.progress_bar import ProgressBarData
+from servicelib.utils import logged_gather
+from settings_library.aws_s3_cli import AwsS3CliSettings
+from simcore_sdk.node_ports_common import aws_s3_cli
+
+pytest_simcore_core_services_selection = [
+    "migration",
+    "postgres",
+    "storage",
+    "redis",
+]
+
+pytest_simcore_ops_services_selection = [
+    "minio",
+    "adminer",
+]
+
+
+WAIT_FOR_S3_BACKEND_TO_UPDATE: Final[float] = 1.0
+
+
+@pytest.fixture
+async def cleanup_bucket_after_test(aws_s3_cli_settings: AwsS3CliSettings) -> None:
+    session = aioboto3.Session(
+        aws_access_key_id=aws_s3_cli_settings.AWS_S3_CLI_S3.S3_ACCESS_KEY,
+        aws_secret_access_key=aws_s3_cli_settings.AWS_S3_CLI_S3.S3_SECRET_KEY,
+    )
+    async with session.resource(
+        "s3",
+        endpoint_url=aws_s3_cli_settings.AWS_S3_CLI_S3.S3_ENDPOINT,
+    ) as s_3:
+        bucket = await s_3.Bucket(aws_s3_cli_settings.AWS_S3_CLI_S3.S3_BUCKET_NAME)
+        s3_objects = [_ async for _ in bucket.objects.all()]
+        await asyncio.gather(*[o.delete() for o in s3_objects])
+
+
+# put to shared config
+def _fake_s3_link(aws_s3_cli_settings: AwsS3CliSettings, s3_object: str) -> AnyUrl:
+    return parse_obj_as(
+        AnyUrl,
+        f"s3://{aws_s3_cli_settings.AWS_S3_CLI_S3.S3_BUCKET_NAME}/{urllib.parse.quote(s3_object)}",
+    )
+
+
+# put to shared config
+async def _create_random_binary_file(
```

Comment:
we also have fixtures for this: see https://github.com/ITISFoundation/osparc-simcore/blob/cdbc9327fdf54f161f64918a1de0afa2f964d2ef/packages/pytest-simcore/src/pytest_simcore/file_extra.py

Reply:
as above